morph_tool/transform.py

```diff
@@ -233,12 +233,14 @@ def align_morphology(
         L.info('We did not find an apical point to align the morphology')
         return np.eye(3)
 
-    principal_direction = _get_principal_direction(
-        points - np.linalg.mean(morph.soma.points, axis=0)
-    )
+    center = np.linalg.mean(morph.soma.points, axis=0)
+
+    principal_direction = _get_principal_direction(points - center)
     principal_direction *= np.sign(points.dot(principal_direction).sum())
 
     rotation_matrix = rotation_matrix_from_vectors(principal_direction, direction)
+    translate(morph, -center)
     rotate(morph, rotation_matrix)
```

Comment:
rotate seems to take origin as an argument. Wouldn't that work instead of doing it manually?